File: Csharp/Creational/Factory/Show/Show/Creator.cs
namespace Show
{
    abstract class Creator
    {
        public abstract Product FactoryMethod();
        public string SomeOperation()
        {
            var product = FactoryMethod();
            var result = "Creator: This creator's code has just worked with "
                         + product.Operation();

            return result;
        }
    }
}

File: Language/JavaScript/Creational/Factory/Example1/Creator.js
// Creator.js
// Purpose: Abstract Creator class declaring factoryMethod()

class Creator {
    factoryMethod() {
        throw new Error("Abstract method 'factoryMethod' must be implemented by subclass.");
    }

    someOperation() {
        const product = this.factoryMethod();
        return `Creator: Working with ${product.operation()}`;
    }
}

export default Creator;
